### DISCUSSION TITLE
How should you add Google maps API to a next app?

### DISCUSSION THREAD

**Discussion by grantspilsbury** [vercel/next.js#37611] [Help] [answered]
I want to use the Google maps api for a shipping address autocomplete. I am using Nextjs 12 and Node v16.4.

@react-google-maps/api does not work.

Add a Script element to _document.js does not work.
```
import { Html, Head, Main, NextScript } from 'next/document'
import Script from 'next/script'

const source = `https://maps.googleapis.com/maps/api/js?key=${process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY}&libraries=places`

const Document = () => {
  return (
    <Html>
      <Head>
        <Script type="text/javascript" src={source} strategy="beforeInteractive" />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}

export default Document
```

google remains undefined.

Any help would be appreciated.

Thanks!

**Comment by icyJoseph** [comment, marked as answer]
Hi, adding `Script` inside `_document` Head has a bug, for now please just use a regular script tag. I'll update with an example.

Not sure why `@react-google-maps/api` does not work for you?

### Edit

Relevant links:

- https://github.com/vercel/next.js/pull/37000
- https://github.com/vercel/next.js/issues/37225#issuecomment-1142390675
- https://github.com/vercel/next.js/pull/37084,

```jsx
<script
  type="text/javascript"
  src="https://unpkg.com/lodash@4.17.20/lodash.js"
></script>
```

Something like this, and when the next version comes out, upgrade to `next/script` directly!

**Reply by IfeanyiAlexander** [reply]
i did this, it worked fine but when running npm run build, it fails with message
 **next/script should not be used in pages/_document.js.**
